File: InventoryManagement.Api/Contracts/BillModel.cs
using System;
using InventoryManagement.Domain.Model;

namespace InventoryManagement.Api.Contracts;

public class BillModel
{
    public int Id {get; set;}
    public string Name {get; set;}
    public string Mobile {get; set;}
    public int? Discount {get; set;}
    public int? Advance {get; set;}
    public string? PaymentMode {get; set;}
    public DateTimeOffset BillDate{get; set;} = DateTimeOffset.Now;
    public  string? Status { get; set;}


}

File: InventoryUI/inventory-ui/src/app/models/bill.model.ts
import { BillItemModel } from "./bill-item.model";

export interface BillModel {
  id?: number;
  name: string;
  mobile?: string;
  discount?: number;
  advance?: number;
  paymentMode?: string;
  billDate?: Date;
}

export interface GetBillModel {
  id?: number;
  name: string;
  mobile?: string;
  discount?: number;
  advance?: number;
  paymentMode: number;
  billDate?: Date;
  paymentUser?: string;
}

export interface GetAllBillModel {
  id?: number;
  name: string;
  mobile?: string;
  discount?: number;
  advance?: number;
  paymentMode: number;
  billDate?: Date;
  paymentUser?: string;
  totalAmount?: number;
}